src/vcy/ast.ml

```diff
@@ -138,6 +138,12 @@ and stmt =
 | Assert of exp node
 | Assume of exp node
 | Havoc of id
+| Require of exp node
+| GCommute of commute_variant * commute_condition * commute_pre_cond * block node list * commute_post_cond
```

Comment:
`| Commute of commute_variant * commute_condition * block node list * commute_pre_cond option * commute_post_cond option`